src/metatrain/utils/evaluate_model.py

```diff
@@ -307,3 +256,50 @@ def _get_model_outputs(
                 for key, value in targets.items()
             },
         )
+
+
+def _prepare_system(system: System, positions_grad: bool, strain_grad: bool):
+    """
+    Prepares a system for gradient calculation.
+    """
+    if strain_grad:
+        strain = torch.eye(
+            3,
+            requires_grad=True,
+            dtype=system.cell.dtype,
+            device=system.cell.device,
+        )
+        new_system = System(
+            positions=system.positions @ strain,
+            cell=system.cell @ strain,
+            types=system.types,
+        )
+    else:
+        if positions_grad:
```

Comment:
I think this is wrong if both `strain_grad` and `positions_grad` are True. The positions would not have gradients.

Reply:
This is correct (at least in the way we use it later). The positions will require grad (because the `strain` from which they're calculated do), and then we call `torch.autograd.grad` (and not `backward` as you might do in the simulation engines) which lets us target the positions tensor as the one that we're differentiating with respect to

Reply:
Fair enough!